ompi/proc/proc.h

```diff
@@ -369,16 +369,42 @@ static inline bool ompi_proc_is_sentinel (ompi_proc_t *proc)
     return (intptr_t) proc & 0x1;
 }
 
-static inline intptr_t ompi_proc_name_to_sentinel (opal_process_name_t name)
+/*
+ * we assume an ompi_proc_t is at least aligned on two bytes,
+ * so if the LSB of a pointer to an ompi_proc_t is 1, we have to handle
+ * this pointer as a sentinel instead of a pointer.
+ * a sentinel can be seen as a 64 bits array with the following format :
+ * - bit  0     : 1
+ * - bits 1-15  : local jobid
+ * - bits 16-31 : job family
+ * - bits 32-63 : vpid
```

Comment:
it helped me understanding where the confusion is coming from :-)
a sentinel is basically an ```opal_process_name_t``` packed/encoded/hidden inside an ```ompi_proc_t *```

here is a snippet of how sentinel are used, from ```ompi/dpm/dpm.c```
```c
        if (ompi_proc_is_sentinel (proct)) {
            proc_name = ompi_proc_sentinel_to_name ((uintptr_t)proct);
        } else {
            proc_name = proct->super.proc_name;
        }
```
```proct``` has type ```ompi_proc_t *```, but it can either be a genuine pointer that can be dereferenced, or a sentinel.
so at first, we must figure out whether this 64 bit thing is a pointer or a sentinel.
since an ```ompi_proc_t``` object is aligned on at least two bytes, it is safe to use the LSB of the 64 bit thing. (if it is zero, then it is a pointer, if it is one, then it cannot be a pointer and it is a sentinel).
if you simply use the MSB of the local jobid, you cannot tell whether the 64 bit thing is a pointer or a sentinel.

using the LSB of the 64 bit thing does require some bit shifting, but we just can't avoid it.

does that make sense now ?